### ISSUE TITLE
Production cannot find font files

### ISSUE THREAD

**Issue by crystalfp** [electron-userland/electron-builder#7786]
* **Electron-Builder Version**:  24.6.4

* **Node Version**: 18.16.1

* **Electron Version**: 26.2.1
* **Electron Type (current, beta, nightly)**: current

* **Target**: Windows 11 64bits

To load a local font I define in one of the loaded style files:
```
@font-face {
  font-family: "Quattrocento Sans";
  font-style: normal;
  font-weight: normal;
  src: url("/fonts/QuattrocentoSans-Regular.woff2") format("woff2");
  font-display: swap;
}
```
The `fonts` directory is under `public` and the application accesses the font with `font-family: "Quattrocento Sans",sans-serif;`.

In **development** everything works and the font is loaded and used. 
In **production** the font file is not found and the font defaults to `sans-serif` (I have put here `monospace` to make more visible the problem).
Investigating, the `public` directory content with the `fonts` directory is copied under `dist` and the transformed `dist/assets/index-a8f324cb.css` file contains my `@font-face` source with url changed to `url(../fonts/QuattrocentoSans-Regular.woff2)` because `fonts` is one leve up respect to the CSS file.

The `electron-builder.yaml` is:
```
---
"$schema": https://raw.githubusercontent.com/electron-userland/electron-builder/master/packages/app-builder-lib/scheme.json
appId: com.electron.app
productName: my-app
asar: true
directories:
  output: release/${version}
files:
- dist-electron
- dist
electronLanguages:
- en-US
- it
mac:
  artifactName: "${productName}_${version}.${ext}"
  target:
  - dmg
win:
  target:
  - target: nsis
    arch:
    - x64
  artifactName: ${productName}-${version}-setup.${ext}
nsis:
  oneClick: false
  perMachine: false
  allowToChangeInstallationDirectory: true
  deleteAppDataOnUninstall: false
  shortcutName: ${productName}
  uninstallDisplayName: ${productName}
  createDesktopShortcut: always
```
What is not working here? How could I check what is contained in the ASAR file?
Thanks!
mario

**Comment by mmaietta** [comment]
>What is not working here? How could I check what is contained in the ASAR file?

To inspect the asar package, you can leverage the official [asar](https://github.com/electron/asar) package and use `asar extract <path to file> <dest folder>` IIRC

**Comment by crystalfp** [comment]
Thanks for the suggestion @mmaietta !
Yes, in the asar file there are: `\dist\fonts\Quattrocento-Regular.woff2` and `\dist\assets\index-a8f324cb.css` so the problem is not in `electron-builder` or `vite`. So, sorry for the noise, but I have to continue my search to find what blocks the access to the font files.
Thanks again!
mario

**Comment by crystalfp** [comment]
FYI I have found what happened.

In `index.html` of the electron project boilerplate I used at the beginning of my learning of Electron there is the following line:
```
<meta
  http-equiv="Content-Security-Policy"
  content="default-src 'self'; script-src 'self'; style-src 'self' 'unsafe-inline'" />
```
I had to modify this line to make a component ([vue3-tags-input](https://www.npmjs.com/package/vue3-tags-input)) that uses `data:` to define a font by appending `; font-src data:` to make it work. But this blocked the loading of font files.

Now I deleted the entire meta statement apparently without avverse effects and the problem is solved.

Thanks for your patience!
mario
